### PR TITLE
Throw UnknownHostException again from HttpURLConnection

### PR DESCRIPTION
**Pull request by nfuller**
This is to address issue #1827.

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 1]
```diff
@@ -584,11 +586,13 @@ public Builder newBuilder() {
   }
 
   /**
-   * Returns a new {@code OkUrl} representing {@code url} if it is a well-formed HTTP or HTTPS URL,
-   * or null if it isn't.
+   * Returns a new {@code HttpUrl} representing {@code url} if it is a well-formed HTTP or HTTPS
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: Thanks!

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 2]
```diff
@@ -599,6 +603,29 @@ public static HttpUrl get(URL url) {
...
   }
 
+  /**
+   * Returns a new {@code HttpUrl} representing {@code url} if it is a well-formed HTTP or HTTPS
+   * URL, or throws an exception if it isn't.
+   *
+   * @throws MalformedURLException if there was a non-host related URL issue
+   * @throws UnknownHostException if the host was invalid
+   */
+  static HttpUrl getChecked(URL url) throws MalformedURLException, UnknownHostException {
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: Mind making this method take a `String`? Then the caller can call `url.toString()`, which potentially makes this slightly more applicable.

**Comment by swankjesse** [comment]
LGTM! one small suggestion.

**Comment by nfuller** [comment]
Made the change: URL is now String.

**Comment by swankjesse** [comment]
Nice!

**Pull request merged**